src/dvr/dvr_db.c

```diff
@@ -57,6 +57,10 @@ static void dvr_timer_stop_recording(void *aux);
 static int dvr_entry_rerecord(dvr_entry_t *de);
 static time_t dvr_entry_get_segment_stop_extra(dvr_entry_t *de);
 
+static dvr_entry_t *_dvr_duplicate_event(dvr_entry_t *de);
+
+int showskipped = 0;
```

Comment:
I had not thought about it, because I'm the only use tvheadend at home, you're right, but my skills is not enought for do it in other way. If you could give me some ideas I would try to solve. I would like to progress in programming, but I'm self learning programmer. Thanks.

Reply:
It seems to me: api_idnode_grid_conf creates a config with a field filter from the http request from api_idnode_grid. So I imagine api_dvr_entry_grid_upcoming (which takes a conf parameter) would apply such a filter. The filters are specified in the idnode.h.

There's some logic in idnode.js for building the filters[].

So perhaps a filter can (somehow) be applied directly to the grid_upcoming in the Javascript code and it would remove the need for any C code at all since you could possibly apply a "duplicate eq 0" filter and it would remove the duplicates.

Then, in the future, we can add a "title regex" search which is what I'd quite like for dvr panels.

But that's based on a quick look through idnode and I imagine the difficult part is forcing the filter in the JS. There is already filter logic in the epg.js that may help, but that uses a slightly different way of presenting information so it may not prove too helpful.

Reply:
Hi, now I able to filter the store data pressing the button on the client side, but I have a little problem, I can't update the "pagingtoolbar" after apply the filter, so it still showing the total count not the filtered count. Do you know how can I do it? Thanks. 

This is the button code:

var showSkippedButton = {
        name: 'showskipped',
        builder: function() {
            return new Ext.Toolbar.Button({
                tooltip: _('Show / hidde duplicate schedules'),
                iconCls: 'showskipped',
                text: _('Skipped'),
                disabled: false
            });
        },
        callback: function(conf, e, store, select) {
                  if (store.isFiltered()){
        	     store.clearFilter();
        	  }else{
        	     store.filter('duplicate', 0);
        	 }
        }
    };

Reply:
Fantastic.

I don't know how to do update the total count. I just checked the EPG and that does update the counts. Its code in epg.js has "totalProperty: 'totalCount'" and elsewhere it has a "epgStore.getTotalCount" when creating an autorec rule which definitely returns the filtered count. So, I don't know if that's useful?

_Maybe_ setting "grid.getStore().totalProperty='totalCount'"?  It looks like it defaults to "total" in idnode.js, so maybe setting it after the create in dvr.js:351?

Let me know if that works, otherwise I probably won't have time to look more before the new year, but I'll take a look then. (And if I don't, please add a new comment to remind me).

Reply:
After some test, the filter only works in the actual page, so now the client request to the server at two differents url's, one filtered and the other one as it.